include/networkit/algebraic/CSRGeneralMatrix.hpp

```diff
@@ -56,11 +56,30 @@ class CSRGeneralMatrix {
     index binarySearchColumns(index left, index right, index j) const {
         assert(sorted());
         const auto it = std::lower_bound(columnIdx.begin() + left, columnIdx.begin() + right, j);
-        if (it == columnIdx.end() || *it != j)
-            return none;
-        return it - columnIdx.begin();
+        if (*it > j) {
+            if (it == columnIdx.begin())
+                return none;
+            return std::prev(it) - columnIdx.begin();
+        } else
+            return it - columnIdx.begin();
```

Comment:
I think this code could be simplified to:
```cpp
if (it == columnIdx.begin())
    return none;
if (*it > j)
    it =  std::prev(it);
return it - columnIdx.begin();
```

Reply:
this has been adressed by removing that function entirely